packages/router-core/src/scroll-restoration.ts

```diff
@@ -316,3 +316,17 @@ export function setupScrollRestoration(router: AnyRouter, force?: boolean) {
     }
   })
 }
+
+export function handleHashScroll(router: AnyRouter) {
```

Comment:
Couple of small changes here please:

- Rename to `handleTransitionerHashScroll`
- Annotate the function with some JSDoc comments marking it as `@internal` with a short description saying that this is to be setup in the `<Transitioner>` component and what it actually does.

Reply:
Optionally, the rename is not necessary if just the JSDoc comment being there.